### DISCUSSION TITLE
baseurl nginx reverse proxy

### DISCUSSION THREAD

**Discussion by userdocs** [drakkan/sftpgo#370] [General]
Hello, I was wondering if there is any kind of baseurl support for this application.

I would like to run it at `/sftpgo` using nginx as a reverse proxy but i just get his message

```
{"message":"Not Found"}
```
It loads the webui fine on `/`

If this app can be run on a subdirectory using nginx how can you do it?

**Comment by drakkan** [comment]
Hi,

I'm not an nginx expert and maybe there is a better way, however something like this works for me:

```
        location /sftpgo/ {
            proxy_pass http://127.0.0.1:8080/;
        }
        location /web/ {
            proxy_pass http://127.0.0.1:8080/web/;
        }
        location /static/ {
            proxy_pass http://127.0.0.1:8080/static/;
        }
```

To have the `sftpgo` prefix for all paths some changes are required

**Reply by userdocs** [reply]
As far as i can see that is just creates a redirect from

```
http://127.0.0.1/sftpgo
```

To 

```
http://127.0.0.1/web/login
http://127.0.0.1/web/users
```

What I would like is to have

```
http://127.0.0.1/sftpgo/web/login
```

**Comment by userdocs** [comment]
If i build sftpgo like this

https://github.com/userdocs/sftpgo/commit/124332bedaaa394a7c4dcfec165dec6f37650634

And change the hardcoded paths in the templates, it works ( still a bad redirect but the main url loads.

https://github.com/userdocs/sftpgo/commit/a6a3a4891b0804e4b3cb48665d7e961722241c7b

i can access by visiting `/sftpgo/web/login`.

`/sftpgo/` will still give an error but that's not a huge issue.

**Reply by drakkan** [reply]
Yes, this is what I meant when I wrote

> To have the sftpgo prefix for all paths some changes are required

but we cannot hardcode this way, this should be configurable and I think it is not required for `/api/v2/...` paths, only for `/web/*` and (maybe) `/static/*`,

**Reply by userdocs** [reply]
A dynamic method to set it would be ideal.  something like `./serve --base-url /sftpgo serve`

Some other `go` apps like filebrowser have a base url option. maybe it can help?

https://github.com/filebrowser/filebrowser/search?q=baseurl

**Reply by drakkan** [reply]
it is configurable [now](https://github.com/drakkan/sftpgo/commit/0bc4db9950fec38fa32b29f820e7e864910bd864). This feature will be included in 2.1 (this new release will take a while).

You can configure the web admin root within `sftpgo.json`, for example:

```
...
"web_admin_root": "/sftpgo"
...
```

Here is a working nginx config:

```
        location /sftpgo/ {
            proxy_pass http://127.0.0.1:8080/sftpgo/;
        }

        location /sftpgo/api/v2/ {
            proxy_pass http://127.0.0.1:8080/api/v2/;
        }

        location /sftpgo/healthz {
            proxy_pass http://127.0.0.1:8080/healthz;
        }
```

you can test using the `edge` docker variants (available in about 30 minutes) or the build artificats available [here](https://github.com/drakkan/sftpgo/actions/runs/734212033). Thank you

**Reply by userdocs** [reply]
Nice one, thank you for that! I will try it out

**Reply by userdocs** [reply]
So a quick install using the build artifact `sftpgo-linux-amd64-go-1.16` worked fine.

I used your nginx config and set the base to `/sftpgo`. No redirect issues or inspector/browser issue so it looks good.

I will test more and let you know if come across any issue. Thanks again.

**Comment by derekmahar** [comment]
[Here](https://github.com/drakkan/sftpgo/issues/579#issuecomment-1147755381) is my SWAG (NGINX) reverse proxy configuration for SFTPGo.

**Comment by biomassboilers** [comment]
Is `web_admin_root` still valid? I'm [having trouble with running SFTPGo behind a reverse proxy](https://github.com/drakkan/sftpgo/discussions/2051) and `web_admin_root` doesn't do anything. `web_root` and `static_files_path` don't work properly, either.
